schemas/json/theme.json

```diff
@@ -533,48 +546,57 @@
 				{ "$ref": "#/definitions/settingsPropertiesCustom" }
 			]
 		},
-		"settingsPropertiesComplete": {
+		"settingsPropertiesNames": {
+			"enum": [
+				"appearanceTools",
+				"border",
+				"color",
+				"layout",
+				"spacing",
+				"typography",
+				"custom"
+			]
+		},
+		"settingsPropertiesNested": {
 			"type": "object",
 			"allOf": [
 				{
 					"$ref": "#/definitions/settingsProperties"
 				},
 				{
-					"properties": {
-						"appearanceTools": {},
-						"border": {},
-						"color": {},
-						"layout": {},
-						"spacing": {},
-						"typography": {},
-						"custom": {}
-					},
-					"additionalProperties": false
+					"$ref": "#/definitions/settingsBlocksPropertiesComplete"
 				}
-			]
+			],
+			"propertyNames": {
+				"anyOf": [
+					{ "$ref": "#/definitions/settingsPropertiesNames" },
+					{ "$ref": "#/definitions/settingsBlocksNames" }
+				]
+			}
 		},
 		"settingsBlocksPropertiesComplete": {
 			"type": "object",
 			"properties": {
 				"core/archives": {
-					"description": "Archive block. Display a monthly archive of your posts. This block has no block-level settings",
-					"additionalProperties": false
+					"description": "Archive block. Display a monthly archive of your posts. This block has no block-level settings"
 				},
 				"core/audio": {
-					"$ref": "#/definitions/settingsPropertiesComplete"
+					"$ref": "#/definitions/settingsPropertiesNested"
```

Comment:
If a block doesn't support inner blocks, should we allow nested properties?

We're avoiding the specific settings that are available to each block because blocks are getting new supports pretty often and there's a high maintenance overhead for keeping the specifics up to date, but blocks updating to support nesting happens pretty rarely, so I don't think it would be too big of a problem to use `settingsPropertiesComplete` for blocks that don't have inner blocks.

Reply:
That's a great idea - limiting which blocks allow nesting via the schema will also make the intellisense a lot more helpful.